### PR TITLE
added class ArrayList with implements List interface

### PR DESCRIPTION
**Pull request by mstefanchuk**

### CODE REVIEW

File: src/main/java/core/basesyntax/ArrayList.java [hunk 1]
```diff
@@ -1,48 +1,106 @@
...
+                }
+            }
+            throw new NoSuchElementException("Element not found");
+        }
+        for (int i = 0; i < size; i++) {
+            if (element.equals(elements[i])) {
+                return remove(i);
+            }
+        }
+        throw new NoSuchElementException("Element not found");
```
Comment (Vasul-Pankiv) [inline src/main/java/core/basesyntax/ArrayList.java:73]: use if-else and throw exception only once after if-else

**Review by Vasul-Pankiv — CHANGES_REQUESTED** [review]
Great Job!

File: src/main/java/core/basesyntax/ArrayList.java [hunk 2]
```diff
@@ -1,48 +1,106 @@
...
+                    return remove(i);
+                }
+            }
+        } else {
+            for (int i = 0; i < size; i++) {
+                if (element.equals(elements[i])) {
+                    return remove(i);
+                }
+            }
+        }
```
Comment (olekskov) [inline src/main/java/core/basesyntax/ArrayList.java:72]: I'd recommend not to split logic for null

**Review by olekskov — APPROVED** [review]
Good job

**Pull request closed without merging**